enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,517 @@
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OPA Configuration"
+  OPA *OPASpec `json:"opa"`
+  // Roles defines a set of permissions to interact with a tenant.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Static Roles"
+  Roles []*RoleSpec `json:"roles"`
+  // RoleBindings defines configuration to bind a set of roles to a set of subjects.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Static Role Bindings"
+  RoleBindings []*RoleBindingsSpec `json:"roleBindings"`
+}
+
+// OPASpec defines the opa configuration spec for lokiStack Gateway component.
+type OPASpec struct {
+  // URL defines the third-party endpoint for authorization.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OpenPolicyAgent URL"
+    URL string  `json:"url"`
+}
+
+// RoleSpec describes a set of permissions to interact with a tenant.
+type RoleSpec struct {
+	Name        string           `json:"name"`
+	Resources   []string         `json:"resources"`
+	Tenants     []string         `json:"tenants"`
+	Permissions []PermissionType `json:"permissions"`
+}
+
+// RoleBindingsSpec binds a set of roles to a set of subjects.
+type RoleBindingsSpec struct {
+	Name     string    `json:"name"`
+	Subjects []Subject `json:"subjects"`
+	Roles    []string  `json:"roles"`
+}
+
+// Subject represents a subject that has been bound to a role.
+type Subject struct {
+	Name string      `json:"name"`
+	Kind SubjectKind `json:"kind"`
+}
+```
+
+## Proposal
+
+### LokiStack Gateway configuration using _Modes_
+
+LokiStack Gateway can operate in 3 different modes based on user's choice.
+
+* Static Mode - It will offer static OPA configuration with a simple default rego file that will be part of the lokiStack gateway deployment as a static file. This static rego file needs to understand the RbacSpec. This makes `RbacSpec` mandatory.
+
+* Dynamic Mode - It will offer static OPA configuration along with an OPA url to delegate authorization of OIDC extracted subject information to a third-party OPA-compatible endpoint. This makes `OPA URL` mandatory and ignores `RbacSpec`.
+
+* Openshift Logging Mode - It will not need any information from the user. The lokiStack gateway deployment will configure dynamic OPA authorization itself using [opa-openshift](https://github.com/observatorium/opa-openshift/pull/1), because the sidecar will handle authorization based on SubjectAccessReviews against the API server.
+
+### LokiStack CR configuration - _Static Mode_
+
+In this mode, user will need to provide tenants, oidc and rbac configuration madatorily in the LokiStack CR. The lokiStack gateway deployment will itself create a static rego file to understand the rbacSpec.
```

Comment:
wording: `In this mode it is mandatory to provide a static list of per-tenant OIDC configuration as well as static RBAC. The LokiStack gateway deployment will be deployed with access to a static Rego file to match the RbacSpec. The static Rego file represents an in-process OPA endpoint.`